vignettes/articles/writing-custom-fns.Rmd

```diff
@@ -0,0 +1,244 @@
+
+The `capture_check_cnd()` function is used to return a check condition and it's output should also be what a custom check function returns in most cases. The function returns a `<hub_check>` class object depending on the value passed to the `check` argument, which represents the summary of the condition being checked by a given validation function. 
+
+If the value passed to `check ` is `TRUE`, the function returns a `<message/check_success>` class object. 
+
+If the value is `FALSE`, the output depends on the `error` argument.  
+
+- If `error` is `TRUE`, the function returns a `<message/check_error>` class object. If `error` is `FALSE`, the function returns a `<error/check_failure>` class object, which indicates the check has failed. 
+- If `error` is `TRUE`, the function returns a `<error/check_error>` class object, which indicates the check has failed and additionally causes execution of further custom validation functions to halt. Set `error = TRUE` if downstream checks cannot be run if the current check fails.
+
+The default value of `error` is `FALSE` in `create_custom_check()` but setting it to `TRUE` sets the error argument to `TRUE`.
+
+```{r}
+create_custom_check("cstm_check_tbl_error",
+  hub_path = hub_path, error = TRUE
+)
+```
+
+```{r, code=readLines(r_path(hub_path, "cstm_check_tbl_error")), eval=FALSE}
+```
+
+
+### Skipping a check
+
+**Returning a check message with `capture_check_info()`**
+
+Sometimes a check function might not always be applicable and a pre-condition needs to be met before the main check itself is performed. **If the pre-condition is not met, the check is usually skipped**. 
+
+**For such checks, the function should return a `<message/check_info>` object**, generated by the `capture_check_info()` function. Use the `msg` argument to explain that a check was skipped and why.
+
+
+```{r}
+capture_check_info(
+  "modelA-teamA/2024-09-12-modelA-teamA",
+  "Condition for running this check was not met. Skipped."
+)
+```
+
+For example, the `check_tbl_value_col_ascending()` check function which validates that values are ascending when arranged by increasing `output_type_id` order is only applicable to `cdf` and `quantile` output types. Before proceeding with the main check, the function first checks whether the model output `tbl` contains data for `cdf` and `quantile` output types. If not, the check is skipped.
+
+```{r}
+check_tbl_value_col_ascending
+```
+
+
+You can add a pre-condition check block of code with argument `conditional = TRUE` when creating the custom check function with `create_custom_check()`.
+
+```{r}
+create_custom_check("cstm_check_tbl_skip",
+  hub_path = hub_path,
+  conditional = TRUE
+)
+```
+
+
+```{r, code=readLines(r_path(hub_path, "cstm_check_tbl_skip")), eval=FALSE}
+```
+
+
+## Loading config files
+
+Many check conditions against information stored in hub configuration files and these are often read in at the start of the custom check function.
```

Comment:
I found this sentence confusing. 
Is "check conditions" the object rendered by function or is it "Many people wants to check (as in validate) conditions against ..." ? if it's the first one, are we missing a verb??

Reply:
```suggestion
Many checks are conditioned against information stored in hub configuration files and these are often read in at the start of the custom check function.
```